Overview.bs

```diff
@@ -527,15 +527,68 @@ The algorithm:
+the font, but should only use it for the rendering of code points, features, and design space that are fully loaded according to
+[[#ift-font-coverage]]. Rendering of all other content should fallback to a different font following normal client fallback logic.
 
 If the error occurred during [$Load patch file$] then, the client may continue trying to extend the font subset
 if there are remaining patches available other than the one(s) that failed to load. In the case of all other errors the client
 must not attempt to further extend the font subset.
 
+Target Subset Definition {#target-subset-definitions}
+-----------------------------------------------------
+
+The [$Extend an Incremental Font Subset$] algorithm takes as an input a target subset definition based on some content that the client
+wants to render. The client may choose to form one single subset definition for the content as a whole and run the extension algorithm
+once. Alternatively, the client may instead break the content up into smaller spans, form a subset definition for each span, and run
+the extension algorithm on each of the smaller subset definitions. Either approach will ultimately produce a font which equivalently
+renders the overall content as long as:
+
+*  Each span of text which generates a subset definition is built from only one or more complete shaping units.
+
+*  Where a shaping unit is a span of text which the client will process together as a single unit during
+    <a href="https://harfbuzz.github.io/what-is-harfbuzz.html#what-is-text-shaping">text shaping</a>.
+
+
+Determining what Content a Font can Render {#ift-font-coverage}
+-----------------------------------------------------------------------
+
+Given some incremental font (whether the initial font or one that has been partially extended) a client may wish to know what content
+that font can render in it's current state. This is of particular importance where the client is looking to determine which portions
+of the text to use fallback fonts for.
+
+The following procedure can be used by a client to check what parts of some content an incremental font can render:
+
+* Split the content up into the shaping units (see [[#target-subset-definitions]]) on which the content will be processed during text
+    shaping.
+
+* For each shaping unit there are two checks:
+
+    * First, if for any code point in the shaping unit there is not a [[open-type/cmap|cmap]] entry for it, or the entry maps to glyph
+        0 then the incremental font does not fully support rendering the shaping unit.
+
+    * Second, compute the corresponding [=font subset definition=] and execute the [$Extend an Incremental Font Subset$] algorithm,
+        stopping at step 6. If the entry list is not empty then the incremental font does not fully support rendering the shaping unit.
+
+In most cases the client may also wish to know what the font can render at a more granular level than a shaping unit. The following
```

Comment:
On tone: I'd prefer making this and the below less, well, emphatic, I guess. Why "most" here instead of "some"? And why "In some rare cases" below? The subject here is a procedure that can be used for temporary rendering when the font is in an complete state relative to a full shaping unit. Let's just lay it out and be clear that even if all the codepoints are present in the narrow sense the overall result may be different and will have to be re-rendered once the font is patched.

On substance: As the implementer of a client, is what I want to know really "what the font can render at a more granular level than a shaping unit"? It seems more like what I want to know is *how* to (temporarily) render content with the font at its current patch level, to the extent I can use the font. So: we've described the codepoint-by-codepoint check, and that tells us what definitely *can't* be rendered (without risking a "blank"). With that information, the process is presumably something like: 

 * Break the shaping-unit up into sub-units at either side of each sequence of unsupported codepoints (if any).
 * Render each sub-unit separately, with unsupported codepoint sequences rendered using the fallback font.
 * Piece the sub-units back together.

And we could note (again) that if there are no unsupported codepoints, the shaping unit can be rendered together but the result may not be the same as if the patched identified by the extension algorithm were loaded.

Maybe this is all implicit, but I feel like we should at least gesture to something along these lines because talking about treating codepoints individually might lead some readers to think that they should be rendered individually, which is not a good approach.

Reply:
> On tone: I'd prefer making this and the below less, well, emphatic, I guess. Why "most" here instead of "some"? And why "In some rare cases" below? The subject here is a procedure that can be used for temporary rendering when the font is in an complete state relative to a full shaping unit. Let's just lay it out and be clear that even if all the codepoints are present in the narrow sense the overall result may be different and will have to be re-rendered once the font is patched.

Sounds good, I'll adjust the wording.

> 
> On substance: As the implementer of a client, is what I want to know really "what the font can render at a more granular level than a shaping unit"? It seems more like what I want to know is _how_ to (temporarily) render content with the font at its current patch level, to the extent I can use the font. So: we've described the codepoint-by-codepoint check, and that tells us what definitely _can't_ be rendered (without risking a "blank"). With that information, the process is presumably something like:
> 
> * Break the shaping-unit up into sub-units at either side of each sequence of unsupported codepoints (if any).
> * Render each sub-unit separately, with unsupported codepoint sequences rendered using the fallback font.
> * Piece the sub-units back together.

Adding this would essentially just be telling clients how to implement font fallback rendering. Which I feel like is out of scope for this spec. Clients should already have an existing strategy for fallback rendering since the problem exists with regular fonts too. All we need to do in this specification is tell them how to detect where they need to fallback. Normally a client could rely on cmap (or other mechanisms) to do this, but with an IFT font cmap doesn't always accurately represent the capabilities so this more complicated check is needed.

It sounds like I should make this more explicit though and mention that this procedure is meant to slot into existing fallback algorithms as an extension to the usual cmap (or whatever the client normal does) mechanism for determining where to fallback.

Reply:
Also I think I'll add a sentence to call out that clients should not rely on cmap being accurate for IFT fonts.

Reply:
I agree that we probably shouldn't be requiring a specific procedure. On the other hand, there's a general goal or principle of shaping "the largest possible subunits" together that I think is objectively correct compared with the alternatives and that would be good to express somehow. Generally speaking the more you break things apart the more likely you are to get differences, so clients should only break things apart to the extent they need to to implement fallback. Or at least, that's how I've come to think about it.